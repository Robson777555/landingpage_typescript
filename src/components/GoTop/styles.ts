import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;