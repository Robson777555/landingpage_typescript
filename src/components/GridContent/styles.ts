// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;