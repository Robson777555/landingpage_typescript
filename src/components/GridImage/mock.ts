interface GridItem {
  altText: string;
  srcImg: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  background: false,
  title: '<b style="font-size: 2rem;">Tec</b>',
  description: 'Conhe√ßa as principais tecnologias usadas neste projeto:',
  grid: [
    {
      altText: 'JavaScript Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/6/6a/JavaScript-logo.png',
    },
    {
      altText: 'React Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
    },
    {
      altText: 'Next.js Logo',
      srcImg: 'https://assets.vercel.com/image/upload/v1662090959/front/nextjs/twitter-card.png',
    },
    {
      altText: 'Node.js Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg',
    },
    {
      altText: 'Git Logo',
      srcImg: 'https://git-scm.com/images/logos/downloads/Git-Icon-1788C.png',
    },
  ],
};

export default mock;