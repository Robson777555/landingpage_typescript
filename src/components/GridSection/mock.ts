interface GridItem {
  title: string;
  description: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  title: '<b style="font-size: 2rem;">Serviço</b>',
  description: 'Conheça as atividades podem ser feitas para melhorar o seu negócio.',
  background: false,
  grid: [
    {
      title: '<b style="font-size: 1.2rem;">Web</b>',
      description: 'Sites e aplicações web personalizadas com as melhores tecnologias do mercado, garantindo performance e segurança para seu negócio.'
    },
    {
      title: '<b style="font-size: 1.2rem;">Análise</b>',
      description: 'Análise especializada para identificar as melhores soluções tecnológicas para suas necessidades específicas.'
    },
    {
      title: '<b style="font-size: 1.2rem;">Otimiza</b>',
      description: 'Melhoramos a performance e experiência do usuário em suas plataformas existentes.'
    },
  ],
};

export default mock;