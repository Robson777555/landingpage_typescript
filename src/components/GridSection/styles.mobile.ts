import styled, { css } from 'styled-components';

export const MobileGridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    width: 100%;
    max-width: 98vw;
    min-width: 0;
    padding: ${theme.spacings.small};
    box-sizing: border-box;

    h2, h3, h4, h5, h6 {
      font-size: 1.1rem;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }
    p, span, div {
      font-size: 1rem;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.3;
    }
  `}
`;
