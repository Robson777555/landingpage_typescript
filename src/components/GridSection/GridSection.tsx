'use client';
import React from 'react';
import * as Styled from './styles';
import { SectionBackground } from '../SectionBackground/SectionBackground';
import { Heading } from '../Heading/Heading';
import { TextComponent } from '../TextComponent/TextComponent';
import { MobileGridElement } from './styles.mobile';
import { useIsMobile } from '../../hooks/useIsMobile';

interface GridItem {
  title: string;
  description: string;
}

interface GridSectionProps {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

export const GridSection = ({
  title,
  description,
  grid,
  background = false,
}: GridSectionProps) => {
  const isMobile = useIsMobile();

  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Heading as="h2" size="huge">
          <span dangerouslySetInnerHTML={{ __html: title }} />
        </Heading>
        <TextComponent>{description}</TextComponent>
        <Styled.Grid>
          {grid.map((el, idx) =>
            isMobile ? (
              <MobileGridElement key={idx}>
                <h3 dangerouslySetInnerHTML={{ __html: el.title }} />
                <p>{el.description}</p>
              </MobileGridElement>
            ) : (
              <Styled.GridElement key={`${el.title}-${idx}`}>
                <Styled.Title as="h3" dangerouslySetInnerHTML={{ __html: el.title }} />
                <TextComponent>{el.description}</TextComponent>
              </Styled.GridElement>
            )
          )}
        </Styled.Grid>
      </Styled.Container>
    </SectionBackground>
  );
};