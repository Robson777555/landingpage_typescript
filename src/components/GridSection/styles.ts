import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
