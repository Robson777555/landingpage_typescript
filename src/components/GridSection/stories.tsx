import React from 'react';
import { GridSection } from './GridSection';
import mock from './mock';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof GridSection> = {
  title: 'GridSection',
  component: GridSection,
  args: mock,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof GridSection>;

export const Default: Story = {
  render: (args) => <GridSection {...args} />,
};

export const WithBackground: Story = {
  args: {
    ...mock,
    background: true,
  },
};