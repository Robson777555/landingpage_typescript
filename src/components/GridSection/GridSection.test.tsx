import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { GridSection } from './GridSection';
import { describe, it, expect, vi } from 'vitest';
import mock from './mock';
import { useIsMobile } from '../../hooks/useIsMobile';

vi.mock('../../hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}));

describe('<GridSection />', () => {
  it('should render title, description and grid items for desktop', () => {
    vi.mocked(useIsMobile).mockReturnValue(false);

    render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} />
      </ThemeProvider>
    );

    expect(screen.getByRole('heading', { name: /Serviço/i })).toBeInTheDocument();
    expect(screen.getByText(mock.description)).toBeInTheDocument();

    const gridTitles = screen.getAllByRole('heading', { level: 3 });
    expect(gridTitles).toHaveLength(mock.grid.length);

    mock.grid.forEach((item) => {
      expect(screen.getByText(item.description)).toBeInTheDocument();
    });
  });

  it('should render mobile version when isMobile is true', () => {
    vi.mocked(useIsMobile).mockReturnValue(true);

    render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} />
      </ThemeProvider>
    );

    expect(screen.getByRole('heading', { name: /Serviço/i })).toBeInTheDocument();
    expect(screen.getByText(mock.description)).toBeInTheDocument();

    const gridTitles = screen.getAllByRole('heading', { level: 3 });
    expect(gridTitles).toHaveLength(mock.grid.length);

    mock.grid.forEach((item) => {
      expect(screen.getByText(item.description)).toBeInTheDocument();
    });
  });

  it('should match snapshot with background for desktop', () => {
    vi.mocked(useIsMobile).mockReturnValue(false);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} background={true} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot for mobile', () => {
    vi.mocked(useIsMobile).mockReturnValue(true);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render HTML tags in titles correctly', () => {
    vi.mocked(useIsMobile).mockReturnValue(false);

    render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} />
      </ThemeProvider>
    );

    const mainTitle = screen.getByRole('heading', { name: /Serviço/i });
    expect(mainTitle.innerHTML).toContain('<b style="font-size: 2rem;">Serviço</b>');

    mock.grid.forEach((item) => {
      const gridTitle = screen.getByRole('heading', {
        name: new RegExp(item.title.replace(/<[^>]+>/g, '').trim()),
      });
      expect(gridTitle.innerHTML).toContain(item.title);
    });
  });
  it ('Snapshot of page', () => {
    vi.mocked(useIsMobile).mockReturnValue(false);

    const { container } = render(
      <ThemeProvider theme={theme}>
        <GridSection {...mock} />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  })
});