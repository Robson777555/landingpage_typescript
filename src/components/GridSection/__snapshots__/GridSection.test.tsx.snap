// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<GridSection /> > Snapshot of page 1`] = `
<div
  class="sc-hoLldG goHkam"
  data-testid="section-background"
>
  <div
    class="sc-jZhnRx kehEgr"
    data-testid="section-container"
  >
    <div
      class="sc-fmZSGO bcrWjV"
    >
      <h2
        class="sc-bXdtCk ksXdHw"
        data-testid="heading"
      >
        <span>
          <b
            style="font-size: 2rem;"
          >
            Serviço
          </b>
        </span>
      </h2>
      <div
        class="sc-hKDTPf jWXUGw"
        data-testid="text-component"
      >
        Conheça as atividades podem ser feitas para melhorar o seu negócio.
      </div>
      <div
        class="sc-gwsNht cbnKgd"
      >
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Web
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Sites e aplicações web personalizadas com as melhores tecnologias do mercado, garantindo performance e segurança para seu negócio.
          </div>
        </div>
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Análise
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Análise especializada para identificar as melhores soluções tecnológicas para suas necessidades específicas.
          </div>
        </div>
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Otimiza
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Melhoramos a performance e experiência do usuário em suas plataformas existentes.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridSection /> > should match snapshot for mobile 1`] = `
<div
  class="sc-hoLldG goHkam"
  data-testid="section-background"
>
  <div
    class="sc-jZhnRx kehEgr"
    data-testid="section-container"
  >
    <div
      class="sc-fmZSGO bcrWjV"
    >
      <h2
        class="sc-bXdtCk ksXdHw"
        data-testid="heading"
      >
        <span>
          <b
            style="font-size: 2rem;"
          >
            Serviço
          </b>
        </span>
      </h2>
      <div
        class="sc-hKDTPf jWXUGw"
        data-testid="text-component"
      >
        Conheça as atividades podem ser feitas para melhorar o seu negócio.
      </div>
      <div
        class="sc-gwsNht cbnKgd"
      >
        <div
          class="sc-hBDmJg kmiTyb"
        >
          <h3>
            <b
              style="font-size: 1.2rem;"
            >
              Web
            </b>
          </h3>
          <p>
            Sites e aplicações web personalizadas com as melhores tecnologias do mercado, garantindo performance e segurança para seu negócio.
          </p>
        </div>
        <div
          class="sc-hBDmJg kmiTyb"
        >
          <h3>
            <b
              style="font-size: 1.2rem;"
            >
              Análise
            </b>
          </h3>
          <p>
            Análise especializada para identificar as melhores soluções tecnológicas para suas necessidades específicas.
          </p>
        </div>
        <div
          class="sc-hBDmJg kmiTyb"
        >
          <h3>
            <b
              style="font-size: 1.2rem;"
            >
              Otimiza
            </b>
          </h3>
          <p>
            Melhoramos a performance e experiência do usuário em suas plataformas existentes.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridSection /> > should match snapshot with background for desktop 1`] = `
<div
  class="sc-hoLldG kOIwJj"
  data-testid="section-background"
>
  <div
    class="sc-jZhnRx kehEgr"
    data-testid="section-container"
  >
    <div
      class="sc-fmZSGO bcrWjV"
    >
      <h2
        class="sc-bXdtCk ksXdHw"
        data-testid="heading"
      >
        <span>
          <b
            style="font-size: 2rem;"
          >
            Serviço
          </b>
        </span>
      </h2>
      <div
        class="sc-hKDTPf jWXUGw"
        data-testid="text-component"
      >
        Conheça as atividades podem ser feitas para melhorar o seu negócio.
      </div>
      <div
        class="sc-gwsNht cbnKgd"
      >
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Web
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Sites e aplicações web personalizadas com as melhores tecnologias do mercado, garantindo performance e segurança para seu negócio.
          </div>
        </div>
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Análise
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Análise especializada para identificar as melhores soluções tecnológicas para suas necessidades específicas.
          </div>
        </div>
        <div
          class="sc-dYOLZc gyUkna"
        >
          <h3
            class="sc-jMpmlX bCuHwk"
          >
            <b
              style="font-size: 1.2rem;"
            >
              Otimiza
            </b>
          </h3>
          <div
            class="sc-hKDTPf jWXUGw"
            data-testid="text-component"
          >
            Melhoramos a performance e experiência do usuário em suas plataformas existentes.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
